lib/src/transaction_builder.dart

```diff
@@ -90,6 +92,25 @@ class TransactionBuilder {
     return _tx.addOutput(scriptPubKey, value);
   }
 
+  int addNullOutput(dynamic data) {
+    var scriptPubKey;
+    if (data is String) {
+      if (data.length <= network.opreturnSize) {
+        scriptPubKey = bscript.compile([OPS['OP_RETURN'], utf8.encode(data)]);
+      } else {
+        throw new ArgumentError('Too much data, max OP_RETURN size is '+network.opreturnSize.toString());
+      }
+    } else if (data is Uint8List) {
+      scriptPubKey = data;
```

Comment:
This is where the data is added directly to the `scriptPubKey` without an OP_RETURN. It is only done when the data is a string. The OP_RETURN can be added after the `data` input type is handled.

Reply:
Fixed.